File: Initials/solve/cpp/InitialsSolver.cpp
#include <fstream>
#include <topcoder>
#include "Initials.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    string name; tc::read(ifs, name);
    ifs.close();

    std::ofstream ofs(argv[2]);
    Initials solver;
    tc::show(ofs, solver.getInitials(name));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: Initials/solve/ruby/InitialsSolver.rb
#! /usr/bin/env ruby

require_relative "Initials"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  name = reader.next(TString)

  result = Initials.new().getInitials(name)
  IO.write(ARGV[1], Writer.new.next(result, TString).to_s)
end

init
main
